jwk.go

```diff
@@ -254,6 +254,25 @@ func (k *JSONWebKey) IsPublic() bool {
 	}
 }
 
+// Public creates JSONWebKey with corresponding publik key if JWK represents asymmetric private key.
+func (k *JSONWebKey) Public() JSONWebKey {
+	if k.IsPublic() {
+		return *k
+	}
+	ret := *k
+	switch key := k.Key.(type) {
+	case *ecdsa.PrivateKey:
+		ret.Key = key.Public()
+	case *rsa.PrivateKey:
+		ret.Key = key.Public()
+	case ed25519.PrivateKey:
+		ret.Key = key.Public()
+	default:
+		return JSONWebKey{} // returning invalid key
```

Comment:
Would prefer if this returned an error, otherwise potentially confusing to callers.

Reply:
Should it return `(JSONWebKey, error)`, (*JSONWebKey, error)` or `*JSONWebKey` with nil on error? I'm rather new to Go, so I'm unsure what's the most "idiomatic" way.